packages/logs/src/domain/logger.ts

```diff
@@ -39,29 +52,53 @@ export class Logger {
   }
 
   @monitored
-  log(message: string, messageContext?: object, status: StatusType = StatusType.info) {
-    this.handleLogStrategy({ message, context: deepClone(messageContext) as Context, status }, this)
+  log(message: string, messageContext?: object, status: StatusType = StatusType.info, error?: Error) {
+    let context: Context
+
+    if (status === StatusType.error || (error !== undefined && error !== null)) {
+      // Always add origin if status is error (backward compatibility - Remove in next major)
+      const errorContext = ({ origin: ErrorSource.LOGGER } as LogsEvent['error'])!
+
+      if (error !== undefined && error !== null) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏we could remove of the `if` imbrication and the repetition of the "hasErrorParameter" condition with something like:

```ts
let errorContext: LogsEvent['error'] | undefined;

if (status === StatusType.error) {
  // Always add origin if status is error (backward compatibility - Remove in next major)
  errorContext = { origin: ErrorSource.LOGGER };
}

if (error !== undefined && error !== null) {
  ...
  errorContext = {
    ...
  }
}

const context = errorContext
  ? combine({ error: errorContext }, messageContext) as Context
  : deepClone(messageContext) as Context
```

Reply:
👌